src/main/java/net/snowflake/ingest/streaming/internal/FlushService.java

```diff
@@ -351,6 +351,16 @@ void distributeFlushTasks() {
         if (!leftoverChannelsDataPerTable.isEmpty()) {
           channelsDataPerTable.addAll(leftoverChannelsDataPerTable);
           leftoverChannelsDataPerTable.clear();
+        } else if (blobData.size()
```

Comment:
Could the logic here combined with https://github.com/snowflakedb/snowflake-ingest-java/blob/3a3cbc8798e2e4eddab1cc672b9d48d557e48d78/src/main/java/net/snowflake/ingest/streaming/internal/FlushService.java#L407? Both are checking whether a separate blob is needed

Reply:
It could, but it could be triggered in the middle of a chunk, which is makes sense for splitting based on chunk/blob size, but not for the newly added check. The only added check only cares about the number of chunks in a blob, so I put the potential break just before it starts a new chunk.